UI/src/view/MassTransitStopForm.js

```diff
@@ -547,13 +658,28 @@
       eventbus.on('assetPropertyValue:changed', function (event) {
         var property = event.propertyData;
 
-        if(event.id && property.publicId === 'tietojen_yllapitaja' && (_.find(property.values, function (value) {return value.propertyValue == '2';}))){
+        // Handle form reload after administrator change
+        if (property.publicId === 'tietojen_yllapitaja' && (_.find(property.values, function (value) {return value.propertyValue == '2';}))) {
+          isAdministratorELYKeskus = true;
+          property.propertyType = "single_choice";
+          renderAssetForm();
+        }
+        if (property.publicId === 'tietojen_yllapitaja' && (_.find(property.values, function (value) {return value.propertyValue != '2';}))) {
```

Comment:
Just put an ELSE for the first IF and it's enough, don't was necessary another if.